internal/kubernetes/operator/datafederation/datafederation.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/mongodb/mongodb-atlas-cli/internal/kubernetes/operator/features"
+	"github.com/mongodb/mongodb-atlas-cli/internal/kubernetes/operator/resources"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	atlasV1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/common"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"go.mongodb.org/atlas-sdk/v20230201004/admin"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	DeletingState = "DELETING"
+	DeletedState  = "DELETED"
+)
+
+func BuildAtlasDataFederation(dataFederationStore store.DataFederationStore, dataFederationName, projectID, projectName, operatorVersion, targetNamespace string, dictionary map[string]string) (*atlasV1.AtlasDataFederation, error) {
+	dataFederation, err := dataFederationStore.DataFederation(projectID, dataFederationName)
+	if err != nil {
+		return nil, err
+	}
+	if !isDataFederationExportable(dataFederation) {
+		return nil, nil
+	}
+	atlasDataFederation := &atlasV1.AtlasDataFederation{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: "atlas.mongodb.com/v1",
+			Kind:       "AtlasDataFederation",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      resources.NormalizeAtlasName(fmt.Sprintf("%s-%s", projectName, dataFederation.GetName()), dictionary),
+			Namespace: targetNamespace,
+			Labels: map[string]string{
+				features.ResourceVersion: operatorVersion,
+			},
+		},
+		Spec: GetDataFederationSpec(dataFederation, targetNamespace, projectName),
+		Status: status.DataFederationStatus{
+			Common: status.Common{
+				Conditions: []status.Condition{},
+			},
+		},
+	}
+	return atlasDataFederation, nil
+}
+
+func isDataFederationExportable(dataFederation *admin.DataLakeTenant) bool {
+	state := dataFederation.GetState()
+	if state == DeletingState || state == DeletedState {
+		return false
+	}
+	return true
+}
+
+func GetDataFederationSpec(dataFederationSpec *admin.DataLakeTenant, targetNamespace, projectName string) atlasV1.DataFederationSpec {
+	return atlasV1.DataFederationSpec{
+		Project:             common.ResourceRefNamespaced{Name: projectName, Namespace: targetNamespace},
+		Name:                dataFederationSpec.GetName(),
+		CloudProviderConfig: GetCloudProviderConfig(dataFederationSpec.CloudProviderConfig),
+		DataProcessRegion:   GetDataProcessRegion(dataFederationSpec.DataProcessRegion),
+		Storage:             GetStorage(dataFederationSpec.Storage),
+	}
+}
+
+func GetCloudProviderConfig(cloudProviderConfig *admin.DataLakeCloudProviderConfig) *atlasV1.CloudProviderConfig {
+	if cloudProviderConfig == nil {
+		return &atlasV1.CloudProviderConfig{}
+	}
+	return &atlasV1.CloudProviderConfig{
+		AWS: &atlasV1.AWSProviderConfig{
+			RoleID:       cloudProviderConfig.Aws.RoleId,
+			TestS3Bucket: cloudProviderConfig.Aws.TestS3Bucket,
+		},
+	}
+}
+
+func GetDataProcessRegion(dataProcessRegion *admin.DataLakeDataProcessRegion) *atlasV1.DataProcessRegion {
```

Comment:
does this need to be exported?
```suggestion
func getDataProcessRegion(dataProcessRegion *admin.DataLakeDataProcessRegion) *atlasV1.DataProcessRegion {
```